homeassistant/components/netatmo/camera.py

```diff
@@ -334,82 +305,91 @@ def _enable_motion_detection(self, enable):
         try:
             if self._localurl:
                 requests.get(
-                    f"{self._localurl}/command/changestatus?status={_BOOL_TO_STATE.get(enable)}",
+                    f"{self._localurl}/command/changestatus?status="
+                    f"{_BOOL_TO_STATE.get(enable)}",
                     timeout=10,
                 )
             elif self._vpnurl:
                 requests.get(
-                    f"{self._vpnurl}/command/changestatus?status={_BOOL_TO_STATE.get(enable)}",
+                    f"{self._vpnurl}/command/changestatus?status="
+                    f"{_BOOL_TO_STATE.get(enable)}",
                     timeout=10,
                     verify=self._verify_ssl,
                 )
             else:
                 _LOGGER.error("Welcome/Presence VPN URL is None")
                 self._data.update()
-                (self._vpnurl, self._localurl) = self._data.camera_data.cameraUrls(
-                    camera=self._camera_name
+                (self._vpnurl, self._localurl) = self._data.camera_data.camera_urls(
+                    cid=self._camera_id
                 )
                 return None
         except requests.exceptions.RequestException as error:
             _LOGGER.error("Welcome/Presence URL changed: %s", error)
             self._data.update()
-            (self._vpnurl, self._localurl) = self._data.camera_data.cameraUrls(
-                camera=self._camera_name
+            (self._vpnurl, self._localurl) = self._data.camera_data.camera_urls(
+                cid=self._camera_id
             )
             return None
         else:
             self.async_schedule_update_ha_state(True)
 
-    # Netatmo Presence specific camera method.
 
-    def set_light_auto(self):
-        """Set flood light in automatic mode."""
-        _LOGGER.debug(
-            "Set the flood light in automatic mode for the camera '%s'", self._name
-        )
-        self._set_light_mode("auto")
+class CameraData:
+    """Get the latest data from Netatmo."""
+
+    def __init__(self, hass, auth):
+        """Initialize the data object."""
+        self._hass = hass
+        self.auth = auth
+        self.camera_data = None
+        self.camera_ids = []
+        self.module_names = []
+        self.camera_type = None
+
+    def get_camera_home_id(self, camera_id):
+        """Return the home id for a given camera id."""
+        for home_id in self.camera_data.cameras:
+            for camera in self.camera_data.cameras[home_id].values():
+                if camera["id"] == camera_id:
+                    return home_id
+        return None
 
-    def set_light_on(self):
-        """Set flood light on."""
-        _LOGGER.debug("Set the flood light on for the camera '%s'", self._name)
-        self._set_light_mode("on")
+    def get_all_camera_ids(self):
```

Comment:
The problem with this function is that you fetch all camera objects but only return IDs. Now the next piece of code will have to fetch the camera again ?